x-pack/plugins/ml/server/models/results_service/results_service.ts

```diff
@@ -708,6 +721,35 @@ export function resultsServiceProvider(mlClient: MlClient, client?: IScopedClust
       finalResults.bucketResults.push([timestamp, eventCount]);
     });
 
+    const { getAnnotations } = annotationServiceProvider(client!);
+    const resp = await getAnnotations({
+      jobIds: [jobId],
+      earliestMs: start,
+      latestMs: end,
+      maxAnnotations: 1000,
+    });
+
+    const annotationResults = resp.annotations[jobId] || [];
+    annotationResults.forEach((annotation) => {
+      const timestamp = Number(annotation?.timestamp);
+      const endTimestamp = Number(annotation?.end_timestamp);
+      if (timestamp === endTimestamp) {
+        finalResults.annotationResultsLine.push({
+          dataValue: timestamp,
+          details: annotation.annotation,
+          header: dateFormatter(timestamp),
+        });
+      } else {
+        finalResults.annotationResultsRect.push({
+          coordinates: {
+            x0: timestamp,
+            x1: endTimestamp,
+          },
+          details: annotation.annotation,
```

Comment:
Could you add the `header` prop in for the `rect` annotations too, so that the start (and end?) times are shown in the header, as you have for the line annotations? That would give them a consistent style with the line ones.

Reply:
`RectAnnotationDatum` doesn't have a prop for that but, if necessary, we can create a custom tooltip.

Reply:
Not needed for this PR, but worth adding to the meta issue I think. The format would then be consistent with the ones in the results pages where we show the start/end times of the annotations.